File: frontend/lib/data/models/confirm_email_request.dart
/// Request model for email confirmation.
class ConfirmEmailRequest {
  const ConfirmEmailRequest({
    required this.userId,
    required this.code,
    this.changedEmail,
  });

  final String userId;
  final String code;
  final String? changedEmail;

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'code': code,
      if (changedEmail != null) 'changedEmail': changedEmail,
    };
  }
}

File: backend/DTOs/ConfirmEmailRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Axle.DTOs;

/// <summary>
/// Request model for email confirmation.
/// </summary>
public class ConfirmEmailRequest
{
    [Required]
    public string UserId { get; set; } = string.Empty;

    [Required]
    public string Code { get; set; } = string.Empty;
}
